staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller_test.go

```diff
@@ -0,0 +1,339 @@
+			},
+			inputPoolName: testPoolName,
+			inputDriverResources: &DriverResources{
+				Pools: map[string]Pool{
+					testPoolName: {
+						Devices: []resourceapi.Device{
+							{
+								Name: testDeviceName,
+							},
+						},
+					},
+				},
+			},
+			expectedResourceSlices: []resourceapi.ResourceSlice{
+				*newExpectResourceSlice(testResourceSlice1, testNode, string(testNodeUID), testDriveName, testPoolName, testDeviceName, false, 1),
+			},
+		},
+		"no-resourceslices-existing": {
+			testNodeUID:    testNodeUID,
+			initialObjects: []runtime.Object{},
+			inputPoolName:  testPoolName,
+			inputDriverResources: &DriverResources{
+				Pools: map[string]Pool{
+					testPoolName: {
+						Devices: []resourceapi.Device{
+							{
+								Name: testDeviceName,
+							},
+						},
+					},
+				},
+			},
+			expectedResourceSlices: []resourceapi.ResourceSlice{
+				*newExpectResourceSlice(testNode+"-"+testDriveName+"-", testNode, string(testNodeUID), testDriveName, testPoolName, testDeviceName, true, 0),
+			},
+		},
+		"single-resourceslice-existing-with-different-resource-pool-generation": {
+			testNodeUID: testNodeUID,
+			initialObjects: []runtime.Object{
+				newResourceSlice(testResourceSlice1, testNode, testDriveName, testPoolName, 2),
+				newResourceSlice(testResourceSlice2, testNode, testDriveName, testPoolName, 1),
+				newResourceSlice(testResourceSlice3, testNode, testDriveName, testPoolName, 1),
+			},
+			inputPoolName: testPoolName,
+			inputDriverResources: &DriverResources{
+				Pools: map[string]Pool{
+					testPoolName: {
+						Generation: 3,
+						Devices: []resourceapi.Device{
+							{
+								Name: testDeviceName,
+							},
+						},
+					},
+				},
+			},
+			expectedResourceSlices: []resourceapi.ResourceSlice{
+				*newExpectResourceSlice(testResourceSlice1, testNode, string(testNodeUID), testDriveName, testPoolName, testDeviceName, false, 3),
+			},
+		},
+		"single-resourceslice-existing-with-mismatching-resource-pool-name": {
+			testNodeUID: testNodeUID,
+			initialObjects: []runtime.Object{
+				newResourceSlice(testResourceSlice1, testNode, testDriveName, testPoolName, 1),
+			},
+			inputPoolName: testPoolName,
+			inputDriverResources: &DriverResources{
+				Pools: map[string]Pool{
+					testPoolName1: {
+						Generation: 1,
+						Devices: []resourceapi.Device{
+							{
+								Name: testDeviceName,
+							},
+						},
+					},
+				},
+			},
+			expectedResourceSlices: nil,
+		},
+	}
+
+	for name, test := range testCases {
+		t.Run(name, func(t *testing.T) {
+			_, ctx := ktesting.NewTestContext(t)
+			// Create kubeClient
+			kubeClient := fake.NewSimpleClientset(test.initialObjects...)
+
+			// Create controller
+			ctrl := setupController(ctx, testNode, test.testNodeUID, testDriveName,
+				kubeClient, test.inputDriverResources)
+
+			err := ctrl.syncPool(ctx, test.inputPoolName)
+			assert.Equal(t, test.expectedError, err, "unexpected error")
```

Comment:
`test.expectedError` is always nil right now, isn't it?

Testing for some specific error might have to be more complicated. Better use `require.NoError` and drop `expectedError`?

Reply:
done